File: spring-mvc/spring-mvc-1/src/main/java/com/springservletthymeleaf/basic/itemservice/domain/item/ItemRequestMapper.java
package com.springservletthymeleaf.basic.itemservice.domain.item;

import org.hibernate.cache.spi.support.AbstractReadWriteAccess;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import com.springservletthymeleaf.basic.itemservice.util.GenericMapper;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface ItemRequestMapper extends GenericMapper<ItemRequest, Item> {
}

File: spring-mvc/spring-mvc-2/src/main/kotlin/com/springthymeleaf/message/ItemRequestMapper.kt
package com.springthymeleaf.message

import org.mapstruct.*

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface ItemRequestMapper {

  fun toDto(entity: Item): ItemRequest
  fun toEntity(dto: ItemRequest): Item
  fun toDtoList(entityList: List<Item>?): List<ItemRequest>?
  fun toEntityList(dtoList: List<ItemRequest>?): List<Item>?

  @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
  fun updateFromDto(dto: ItemRequest, @MappingTarget entity: Item)
}